app/controllers/users_controller.rb

```diff
@@ -47,22 +47,30 @@ def create
   end
 
   def update
+    @password_verification = user_verification_params
     @user = current_user
     @user = User.find_by(username: params[:id]) if params[:id] && current_user && current_user.role == "admin"
-    @user.attributes = user_params
-    @user.save({}) do |result|
-      if result
-        if session[:openid_return_to] # for openid login, redirects back to openid auth process
-          return_to = session[:openid_return_to]
-          session[:openid_return_to] = nil
-          redirect_to return_to
+    if @user.valid_password?(user_verification_params["current_password"]) || user_verification_params["ui_update"] != "0"
```

Comment:
So here, it looks like if someone submits this form with `ui_update` manually set to anything except `"0"`, they can bypass the password verification. I was wondering if we want to change this to be more secure, since we're going through the trouble to verify in the first place? Thank you!!!

Reply:
@jywarren I changed it to false to make it more secure

Reply:
I guess I will have to change it to nil

Reply:
done

Reply:
done